src/SDKs/Search/DataPlane/Search.Tests/Utilities/SearchAssert.cs

```diff
@@ -34,5 +55,25 @@ private static CloudException ThrowsCloudException(Action action, Action<CloudEx
 
             return e;
         }
+
+        /// <summary>
+        /// A mock SynchronizationContext that can report whether or not it was called by async tasks.
+        /// </summary>
+        private class MockSynchronizationContext : SynchronizationContext
+        {
+            public bool WasCalled { get; set; }
```

Comment:
At the time I wrote it, I wasn't sure if I might want to re-use the same mock more than once in the same test, in which case this needed to be mutable. Things didn't turn out that way. Generally though it's fine to use mutable state in a state-based test (since you're already dealing with mutable state; in this case, the synchronization context's queue).